bool part_analyzed
bool part1
bool part2
bool part3